cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -590,5 +592,57 @@ struct duration_scalar : chrono_scalar<T> {
    */
   typename T::rep count() { return this->value().count(); }
 };
+
+/**
+ * @brief An owning class to represent a list value in device memory
+ */
+class list_scalar : public scalar {
+ public:
+  using value_type = cudf::list_view;
+
+  list_scalar() : scalar(data_type(type_id::LIST)) {}
+  ~list_scalar()                        = default;
+  list_scalar(list_scalar&& other)      = default;
+  list_scalar(list_scalar const& other) = default;
+  list_scalar& operator=(list_scalar const& other) = delete;
+  list_scalar& operator=(list_scalar&& other) = delete;
+
+  /**
+   * @brief Construct a new list scalar object from existing device data
+   *
+   * @param[in] value The value of the list
+   * @param[in] is_valid Whether the value held by the scalar is valid
+   * @param[in] stream CUDA stream used for device memory operations.
+   * @param[in] mr Device memory resource to use for device memory allocation
+   */
+  list_scalar(cudf::column_view const& value,
+              bool is_valid                       = true,
```

Comment:
I believe that @jrhemstad detests boolean parameter 😄  . You should pass in an enum like `enum class Validity { Valid, Invalid};` instead.

Reply:
I think is_valid is the one case where a bool is clearer... Also, the existing scalar `is_valid()` method returns `bool`, not some enum value. @jrhemstad convince me. :)

Reply:
Also, the base class `scalar` has `set_valid`, which takes a `bool` parameter.

Reply:
With the benefit of hindsight, I'd support replacing the `bool is_valid` with an enum. I think

```
auto s = scalar(42, true);
```
is less informative than something like:
```
auto s = scalar(42, is_valid::yes);
```

But such a change isn't the responsibility of this PR.